source/common/grpc/google_async_client_cache.cc

```diff
@@ -0,0 +1,22 @@
+#include "common/grpc/google_async_client_cache.h"
+
+namespace Envoy {
+namespace Grpc {
+
+void AsyncClientCache::init(const ::envoy::config::core::v3::GrpcService& grpc_proto_config) {
```

Comment:
so IIUC, this cache will be re-created every time the config is reloaded. Maybe this is OK.

@htuch @hennna how does this work with the control-plane in practice? Will we wind up rebuilding the config whenever there is a control-plane update?

If so we may need to throw this cache into a longer-lived context (and also expire old entries I think).

Reply:
Sorry why did we stop using the singleton manager? The entire point of the singleton manager is to avoid situations like this?

/wait-any